File: nodejs/src/resources/forex.ts
import { BaseResource } from './base';
import type { ForexItem } from '../types';

export class ForexResource extends BaseResource {
  /**
   * Tüm döviz kurlarını listeler.
   */
  async list(): Promise<ForexItem[]> {
    return this._get<ForexItem[]>('/forex');
  }

  /**
   * Belirli bir döviz kurunun verilerini getirir.
   * @param symbol Döviz çifti kodu (Örn: 'USDTRY', 'EURTRY')
   */
  async get(symbol: string): Promise<ForexItem> {
    if (!symbol || typeof symbol !== 'string') {
      throw new Error('symbol parametresi geçerli bir metin olmalıdır.');
    }
    return this._get<ForexItem>(`/forex/${symbol.trim().toUpperCase()}`);
  }
}

File: python/burkut/resources/forex.py
"""
Forex (Currency Rates) Resource
"""

from typing import Any, Dict, List
from burkut.resources.base import BaseResource


class ForexResource(BaseResource):
    """Canlı döviz kurları (USDTRY, EURTRY, GBPTRY vb.)."""

    def list(self) -> List[Dict[str, Any]]:
        """
        Tüm döviz kurlarını listeler.

        Returns:
            Döviz kurları listesi.
        """
        res = self._get("/forex")
        return res if isinstance(res, list) else []

    def get(self, symbol: str) -> Dict[str, Any]:
        """
        Belirli bir döviz kurunun verilerini getirir.

        Args:
            symbol: Döviz çifti kodu (Örn: 'USDTRY', 'EURTRY', 'GBPTRY')
        """
        if not symbol or not isinstance(symbol, str):
            raise ValueError("symbol parametresi boş olamaz.")
        return self._get(f"/forex/{symbol.strip().upper()}")
